amd/src/mod_hsuforum_save_draft.js

```diff
@@ -0,0 +1,128 @@
+                // This is to restore the draft when the page loads and programatic click occurs
+                let pageloadclick = false;
+
+                // Bind click to the reply
+                $('.hsuforum-reply-link').on('click', function() {
+                    window.drafttext = '';
+
+                    let parent = $(this).closest('.hsuforum-post-wrapper');
+                    let postid = $(parent).attr('data-postid');
+
+                    fetchAutoSave(forumid, discussionid, postid, userid);
+
+                    setTimeout(function() {
+                        $(parent).find('#hiddenadvancededitoreditable').html(window.drafttext);
+                    }, 1000);
+                });
+
+                $('.hsuforum-footer-reply .hsuforum-use-advanced').not('.hideadvancededitor').on('click', function(e) {
+                    // Because we toggle editors programmatically depending on how the users interact with the forum, we
+                    // need to differentiate between programmatic clicks after the initial page load click and genuine
+                    // user interactions. If programmatic after page load, prevent any further execution
+                    if (!e.hasOwnProperty('originalEvent') && pageloadclick) {
+                        e.preventDefault();
+                        return false;
+                    }
+
+                    window.drafttext = '';
+                    let postid = null;
+                    pageloadclick = true;
+
+                    fetchAutoSave(forumid, discussionid, postid, userid);
+
+                    setTimeout(function() {
+                        $('.hsuforum-footer-reply').find('#hiddenadvancededitoreditable').html(window.drafttext);
+                    }, 1000);
+                });
+
+
+                // Bind click to new discussion topic button
+                $('#newdiscussionform input[type="submit"]').on('click', function() {
+                    window.drafttext = '';
+
+                    let postid = null;
+
+                    fetchAutoSave(forumid, discussionid, postid, userid);
+
+                    setTimeout(function() {
+                        $('.hsuforum-add-discussion-target').find('#hiddenadvancededitoreditable').html(window.drafttext);
+                    }, 1000);
+                });
+
+                // Bind to keyup to trigger autosave
+                $(document).on("keyup", ".editor_atto_content", function() {
+                    let parent = $(this).closest('.hsuforum-post-wrapper');
+                    let postid = $(parent).attr('data-postid') || null;
+                    let text = $(this).text();
+
+                    if (timer) {
+                        clearTimeout(timer);
+                    }
+
+                    timer = setTimeout(function() {
+                       autoSave(forumid, discussionid, postid, text, userid);
+                    }, 5000); // Wait 5000 milliseconds before triggering event.
```

Comment:
Should we have this as a setting?

Reply:
That is tied to how long to wait after the key-up event before saving the draft. It differs from the normal atto where we specify draft save frequency. More than happy to convert it to that way, was just building off on what Rumano implemented